irispy/sji.py

```diff
@@ -2,880 +2,550 @@
-        """
-        Calculate the standard deviation of the data array.
-        """
-        return SJIMap(np.std(self.data, axis=0), self._meta[self.ref_index])
-
-    def mean(self):
-        """
-        Calculate the mean of the data array.
-        """
-        return SJIMap(np.mean(self.data, axis=0), self._meta[self.ref_index])
-
-    def min(self):
-        """
-        Calculate the minimum value of the data array.
-        """
-        return SJIMap(np.min(self.data, axis=0), self._meta[self.ref_index])
-
-    def max(self):
-        """
-        Calculate the maximum value of the data array.
-        """
-        return SJIMap(np.max(self.data, axis=0), self._meta[self.ref_index])
-
-    def plot(self, axes=None, resample=None, annotate=True, interval=200,
-             plot_function=None, **kwargs):
-        """
-        A animation plotting routine that animates each element in the
-        MapCube
-
-        Parameters
-        ----------
-        gamma: float
-            Gamma value to use for the color map
-
-        axes: mpl axes
-            axes to plot the animation on, if none uses current axes
-
-        resample: list or False
-            Draws the map at a lower resolution to increase the speed of
-            animation. Specify a list as a fraction i.e. [0.25, 0.25] to
-            plot at 1/4 resolution.
-            [Note: this will only work where the map arrays are the same size]
-
-        annotate: bool
-            Annotate the figure with scale and titles
+            # If undo kwarg is NOT set, unmask dust pixels.
+            self.mask[dust_mask] = True
+            self.dust_masked = True
 
-        interval: int
-            Animation interval in ms
 
-        plot_function : function
-            A function to be called as each map is plotted. Any variables
-            returned from the function will have their ``remove()`` method called
-            at the start of the next frame so that they are removed from the plot.
+class IRISMapCubeSequence(NDCubeSequence):
+    """Class for holding, slicing and plotting IRIS SJI data.
 
-        Examples
-        --------
-        >>> import matplotlib.pyplot as plt
-        >>> import matplotlib.animation as animation
-        >>> from sunpy.map import Map
+    This class contains all the functionality of its super class with
+    some additional functionalities.
 
-        >>> cube = Map(files, cube=True)   # doctest: +SKIP
-        >>> ani = cube.plot(colorbar=True)   # doctest: +SKIP
-        >>> plt.show()   # doctest: +SKIP
-
-        Plot the map at 1/2 original resolution
-
-        >>> cube = Map(files, cube=True)   # doctest: +SKIP
-        >>> ani = cube.plot(resample=[0.5, 0.5], colorbar=True)   # doctest: +SKIP
-        >>> plt.show()   # doctest: +SKIP
+    Parameters
+    ----------
+    data_list: `list`
+        List of `IRISMapCube` objects from the same OBS ID.
+        Must also contain the 'detector type' in its meta attribute.
```

Comment:
I think this line is not relevant in the new IRISMapsCubeSequence, what do you think ?